internal/extramanifest/extramanifest.go

```diff
@@ -264,24 +265,30 @@ func (h *EMHandler) ExportExtraManifestToDir(ctx context.Context, extraManifestC
 		return fmt.Errorf("failed to extract manifests from configMap: %w", err)
 	}
 
-	// TODO: sort/group manifests by apply-wave
+	sortedManifests, err := common.SortAndGroupByApplyWave[*unstructured.Unstructured](manifests)
+	if err != nil {
+		return fmt.Errorf("failed to sort and group manifests: %w", err)
+	}
 
 	// Create the directory for the extra manifests
 	exMDirPath := filepath.Join(toDir, ExtraManifestPath)
 	if err := os.MkdirAll(exMDirPath, 0o700); err != nil {
 		return fmt.Errorf("failed to create directory for extra manifests in %s: %w", exMDirPath, err)
 	}
+	for i, manifestGroup := range sortedManifests {
+		group := filepath.Join(toDir, ExtraManifestPath, "group"+strconv.Itoa(i+1))
+		if err := os.MkdirAll(group, 0o700); err != nil {
+			return fmt.Errorf("failed make dir in %s: %w", group, err)
+		}
 
-	for i, manifest := range manifests {
-		gvk := manifest.GroupVersionKind()
-
-		fileName := fmt.Sprintf("%d_%s_%s_%s.yaml", i, gvk.Kind, manifest.GetName(), manifest.GetNamespace())
-		filePath := filepath.Join(toDir, ExtraManifestPath, fileName)
-		err = utils.MarshalToYamlFile(manifest, filePath)
-		if err != nil {
-			return fmt.Errorf("failed to marshal manifest %s to yaml: %w", manifest.GetName(), err)
+		for j, manifest := range manifestGroup {
+			fileName := fmt.Sprintf("%d_%s_%s.yaml", j+1, manifest.GetName(), manifest.GetNamespace())
```

Comment:
gvk.Kind is removed from filename. Is it intentional?

Reply:
Probably not a problem now as it's sorted file starting with number. It was added to ensure unique filename. But I still prefer to add it which helps to know the resource type by just looking at the name. The linux file length limit is long enough, so we don't need to worry about.

Reply:
updated